### PR TITLE
add new dataset

### PR DESCRIPTION
**Pull request by jedrzejkozal**
Add fashion mnsit dataset. It is an implementation of label drift - the idea is that the user preferences change over time, and something that was fashionable is no longer wanted and the other way around.

### CODE REVIEW

File: datasets/seq_cifar100.py
Comment (AlifAshrafee) [inline datasets/seq_cifar100.py:1]: Line 99: `self.classes = list(range(10))` should be `self.classes = list(range(100))`.

Line 166: We currently only use the `train_drift` parameter; the `concept_drift` parameter is unused. We need to change this to apply two different drifts to new training and previously seen data. If we don't use the idea of two different drifts, we can discard the `train_drift` parameter and keep the `concept_drift` parameter.

File: datasets/seq_fashionmnist.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        self.classes = classes_list
+
+    def apply_drift(self, classes: list):
+        if len(set(self.classes).union(classes)) == 0:
+            return
+
+        # switch meta-class in target_transform dictionary
+        for c in classes:
+            self.target_transform[c] = - (self.target_transform[c] - 1)
+
```
Comment (AlifAshrafee) [inline datasets/seq_fashionmnist.py:118]: This formula is such an ingenious way to flip the labels. Love it!

**Review by AlifAshrafee — COMMENTED** [review]
Everything mostly looks good except for a couple of minor issues. I am going to start implementing the label drift in seq_cifar100. I may have found some meaningful meta-classes that might make sense from a practical point of view.

File: datasets/seq_cifar100.py
Comment (jedrzejkozal) [reply datasets/seq_cifar100.py:1]: I've corrected the class range.

For the train_drift argument, I've implemented an additional drift transform: Identity. It doesn't apply any transform to the image. I made it the default train_transform, so now train transforms are disabled by default, but if you want, you can enable them by changing the train_drift argument. Now we can keep this argument and use it if we need it or disable it.

**Pull request merged**